core/src/main/java/org/apache/metamodel/DeleteAndInsertBuilder.java

```diff
@@ -103,7 +103,10 @@ private Row update(final Row original) {
         final DataContext dc = _updateCallback.getDataContext();
         final Table table = getTable();
         final List<FilterItem> whereItems = getWhereItems();
-        final DataSet dataSet = dc.query().from(table).select(table.getColumns()).where(whereItems).execute();
+        final DataSet dataSet = dc.query()
+                .from(table)
+                .select(table.getColumns().toArray(new Column[table.getColumns().size()]))
```

Comment:
right, at first i did not want to change the query interface, but latter added during the refactor, because it was really missing.

Updated it